src/mm/buddy/checkpoint.c

```diff
@@ -1,16 +1,28 @@
 /**
- * @file mm/buddy/ckpt.c
+ * @file mm/buddy/checkpoint.c
  *
- * @brief Checkpointing capabilities
+ * @brief Buddy system checkpointing capabilities
  *
  * SPDX-FileCopyrightText: 2008-2025 HPCS Group <rootsim@googlegroups.com>
  * SPDX-License-Identifier: GPL-3.0-only
  */
-#include <mm/buddy/ckpt.h>
+#include <mm/buddy/checkpoint.h>
 
 #include <core/core.h>
 
 
+/**
+ * @brief Traverses the buddy tree and performs an action on each unallocated block.
```

Comment:
This macro will actually visit allocated blocks (it is used for checkpointing and restore)